### PR TITLE
Add collection update spec

### PR DESCRIPTION
**Pull request by jhmegorei**
When working on the copy_reaction_spec we discovered some code that was better suited to be a proper test case for the collection management spec. This PR adds the test case in the right place and refactors some React components to make JS testing better readable

### CODE REVIEW

File: app/packs/src/apps/mydb/collections/MyCollections.js [hunk 1]
```diff
@@ -114,19 +116,31 @@ export default class MyCollections extends React.Component {
...
+          {this.addCollectionButton(node)}
         </div>
       )
     } else {
       return (
         <ButtonGroup className="actions">
-          <Button id="sync-users-btn" bsSize="xsmall" bsStyle="primary" disabled={node.isNew === true}
-            onClick={() => this.doSync(node, 'CreateSync')}>
+          <Button
+            id="sync-users-btn"
```
[the remark this answers is not in the document]
Comment (jhmegorei) [reply app/packs/src/apps/mydb/collections/MyCollections.js:127]: I'm not sure what you mean, did you comment the right line? In this place, I did not change the ID at all.
I only reformatted the button to make it more readable

File: app/packs/src/apps/mydb/collections/MyCollections.js [hunk 2]
```diff
@@ -114,19 +116,31 @@ export default class MyCollections extends React.Component {
...
+          {isChange && <Button id="save-collections-button" bsSize="xsmall" bsStyle="warning" onClick={this.bulkUpdate.bind(this)}> Save </Button>}
+          {this.addCollectionButton(node)}
         </div>
       )
     } else {
       return (
         <ButtonGroup className="actions">
-          <Button id="sync-users-btn" bsSize="xsmall" bsStyle="primary" disabled={node.isNew === true}
-            onClick={() => this.doSync(node, 'CreateSync')}>
+          <Button
```
[the remark this answers is not in the document]
Comment (jhmegorei) [reply app/packs/src/apps/mydb/collections/MyCollections.js:126]: I don't want to remove it as the multi line version is way easier to read

File: app/packs/src/apps/mydb/collections/MyCollections.js [hunk 3]
```diff
@@ -114,19 +116,31 @@ export default class MyCollections extends React.Component {
...
+            disabled={node.isNew === true}
+            onClick={() => this.doSync(node, 'CreateSync')}
+          >
             <i className="fa fa-plus"></i> <i className="fa fa-share-alt"></i>
           </Button>
-          {this.addButton(node)}
-          <Button bsSize="xsmall" bsStyle="danger" onClick={this.deleteCollection.bind(this, node)}>
+
+          {this.addSubcollectionButton(node)}
+
```
[the remark this answers is not in the document]
Comment (jhmegorei) [reply app/packs/src/apps/mydb/collections/MyCollections.js:137]: This is intentional as the interpolated code (`{this.addSubCollectionButton(node)}` is very easy to miss when wedged between two other components. The new lines make it stand out more, so the eye can discern the individual components better

File: app/packs/src/apps/mydb/collections/MyCollections.js [hunk 4]
```diff
@@ -183,9 +197,27 @@ export default class MyCollections extends React.Component {
...
+      >
+        <i className="fa fa-plus"></i>
+      </Button>
+    )
+  }
+
+  addSubcollectionButton(node) {
     return (
-      <Button id={`mycol_${node.id}`} bsSize="xsmall" bsStyle="success" onClick={this.addSubcollection.bind(this, node)}>
+      <Button
```
[the remark this answers is not in the document]
Comment (jhmegorei) [reply app/packs/src/apps/mydb/collections/MyCollections.js:215]: If by extra line you mean the multiline definition of the component: yes, that is intentional to improve readability :)

**Review by matt-hh — APPROVED** [review]

**Review by mehreenmansur — APPROVED** [review]

**Pull request merged**